pubtools/pulplib/_impl/hooks.py

```diff
@@ -6,14 +6,40 @@
 # pylint: disable=unused-argument
 
 
+@hookspec
+def pulp_repository_pre_publish(repository, options):
+    """Invoked as the first step in publishing a Pulp repository.
+
+    If a hookimpl returns a non-None value, that value will be used to replace
+    the options for this publish. This can be used to adjust publish options
+    from within a hook.
+
+    This should be called before distributors are added, as their config is
```

Comment:
I think this sentence about "should be called" is referring to how this hook is meant to be invoked from within pubtools-pulplib, right?

Could we please drop this sentence?  I think the doc strings on the hooks should be targeted only at the hook implementers, not the callers.  For a hook implementer this part of the doc is just confusing, as they are not supposed to be calling this hook ever.  "before distributors are added" is also vague (added to what?)